src/main/kotlin/app/Cache.kt

```diff
@@ -25,7 +25,10 @@ object Cache {
 
     fun getUserProfile(username: String) = userProfiles[username.toLowerCase()]
     fun contains(username: String) = userProfiles[username.toLowerCase()] != null
-    fun invalid(username: String) = Date().time - (userProfiles[username.toLowerCase()]?.timeStamp ?: 0) > TimeUnit.MILLISECONDS.convert(1, TimeUnit.DAYS)
+    fun invalid(username: String): Boolean = userProfiles[username.toLowerCase()]?.timeStamp
```

Comment:
This block can be cleaned up a bit, but I'm not sure focusing on using `Duration` is the right thing to do. How about:

```
fun getUserProfile(username: String) = userProfiles[username.toLowerCase()]
fun contains(username: String) = getUserProfile(username) != null
fun invalid(username: String) = getUserProfile(username)?.let { Date().time - it.timeStamp > TimeUnit.DAYS.toMillis(1) } ?: false
```

I think 

`Duration.between(Instant.ofEpochMilli(it), Instant.now()).minusDays(1).nano > 0`

is less clear than

`Date().time - it.timeStamp > TimeUnit.DAYS.toMillis(1)`

Reply:
`getUserProfile(username)?.let { HOURS.between(Instant.ofEpochMilli(it.timeStamp), Instant.now()) > 24 } ?: false` could also work?

Reply:
I think the approach of comparing smaller time units is preferable as otherwise the cache would be valid for 25hrs instead of the intended 24.

Reply:
The 24 hour interval was chosen randomly. Heroku restarts the dyno every 24 hours anyway, so this will really never return false for the hosted version (I didn't know that when I decided on 24 hours). You can decrease it to 6 hours, 24 is probably unnecessarily strict.

Reply:
Gotcha - adjusted in my last commit